src/components/Map.tsx

```diff
@@ -0,0 +1,114 @@
+import React, {useEffect, useRef} from 'react';
+import { connect } from "react-redux";
+
+import './Map.css';
+import { State } from "../redux/reducers/reducers";
+
+import {LocationState} from "../redux/reducers/location";
+import {PizzaPlaceState} from "../redux/reducers/pizzaPlaces";
+
+import {getCurrentLocation, setLocation} from "../redux/actions/location";
+import {getPizzaPlaces} from "../redux/actions/pizzaPlaces";
+
+import store from "../redux/store";
+import * as L from 'leaflet';
+import classNames from "classnames";
+import { PizzaPlace } from "../redux/models";
+
+
+interface StateProps {
+    readonly location: LocationState;
+    readonly pizzaPlaces: PizzaPlaceState;
+}
+
+const App = ({location, pizzaPlaces}: StateProps) => {
+    const mapRef: any = useRef(null);
+    const markerRef: any = useRef(null);
+
+    // Get current location once on load
+    useEffect(() => {
+      store.dispatch(getCurrentLocation())
+      store.dispatch(getPizzaPlaces())
+  }, [])
+
+    // Initialize the map on first load
+    useEffect(() => {
+        function onMapClick (d: any) {
+            const latlng = {
+                latitude: d.latlng.lat,
+                longitude: d.latlng.lng
+            }
+            store.dispatch(setLocation(latlng))
+        }
+        mapRef.current = L.map('mapid');
+        L.tileLayer('http://stamen-tiles-{s}.a.ssl.fastly.net/toner-background/{z}/{x}/{y}.png', {
+            attribution: 'Map tiles by <a href="http://stamen.com">Stamen Design</a>, <a href="http://creativecommons.org/licenses/by/3.0">CC BY 3.0</a> &mdash; Map data &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>',
+            maxZoom: 18}).addTo(mapRef.current);
+        mapRef.current.on('click', onMapClick);
+    }, [])
+
+    // Add marker to map whenever location changes
+    useEffect(() => {
+        if (!("isPending" in location || "errorMessage" in location)) {
+            if (!markerRef.current) {
+                markerRef.current = L.circle([location.resource.latitude, location.resource.longitude], {
+                    color: 'blue',
+                    fillColor: '#f03',
+                    fillOpacity: 0.5,
+                    radius: 100
+                }).addTo(mapRef.current);
+            } else {
+                markerRef.current.setLatLng([location.resource.latitude, location.resource.longitude])
+            }
+            markerRef.current.bindPopup("Hey, that's me!")
+            mapRef.current.setView([location.resource.latitude, location.resource.longitude], 13);
+        }
+    }, [location])
+
+    // Add pizza places to map
+    useEffect(() => {
+      console.log("running")
```

Comment:
Not a big deal for a learning project but on client projects we try to be pretty rigorous about not having any `console.log()` calls in shipped code so there will often be linting rules that catch this, or it'll get flagged during review. Same comment for a few lines down.